app/src/main/java/com/example/concordia_campus_guide/Activities/MainActivity.java

```diff
@@ -422,14 +423,13 @@ public Location getMyCurrentLocation() {
     @Override
     public boolean onNavigationItemSelected(@NonNull MenuItem item) {
         int itemId = item.getItemId();
-        SharedPreferences sharedPreferences = getSharedPreferences(ClassConstants.SHARED_PREFERENCES, MODE_PRIVATE);
         if (itemId == R.id.nav_calendar) {
-            SharedPreferences.Editor myEdit = sharedPreferences.edit();
-
-            String value = item.isEnabled() ? ClassConstants.TRUE : ClassConstants.FALSE;
-            myEdit.putString(ClassConstants.CALENDAR_INTEGRATION_BUTTON, value);
-
-            myEdit.commit();
+            if(!calendarViewModel.hasReadPermission() && !calendarViewModel.hasWritePermission()){
+                calendarViewModel.askForPermission(this);
+            }
+            else {
+                Toast.makeText(MainActivity.this, "You have already given permission", Toast.LENGTH_SHORT).show();
```

Comment:
do you actually need to tell them this?

Reply:
well, so that the user wouldn't get confused if they press on the button and nothing happens